### PR TITLE
Add input mouse handling

### PR DESCRIPTION
**Pull request by rstm-sf**
Hello!

Add mouse listener for `MouseEvent`:
- `PointerReleasedEventMessage` will work correctly with `>= Avalonia-0.10.0-preview2` (a backport request was created https://github.com/AvaloniaUI/Avalonia/issues/4579)
- I have not yet found how to implement vertical scrolling
- Also, I have no idea how adding text input and keypress

Fixes parts of  #63

### CODE REVIEW

**Comment by rstm-sf** [comment]
I don’t know if we should use `rd` here? It's a puzzle for me

**Comment by ForNeVeR** [comment]
Wow! This is interesting.

**Comment by rstm-sf** [comment]
I tried using `rd` as I figured it out. We can still try to take out the `JLabel`, but then we have to transfer from one data to another in order to translate into relative coordinates

**Comment by ForNeVeR** [comment]
@rstm-sf, which version of Avalonia is required to test the changes?

File: src/rider/main/kotlin/previewer/AvaloniaMessageMouseListener.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    private fun MouseEvent.avaloniaMouseButton(): Int {
+        return when (this.button) {
+            MouseEvent.BUTTON1 -> MouseButton.Left.ordinal
+            MouseEvent.BUTTON2 -> MouseButton.Middle.ordinal
+            MouseEvent.BUTTON3 -> MouseButton.Right.ordinal
+            else -> MouseButton.None.ordinal
+        }
+    }
+
+    private fun JLabel.relFirstOrNull(first: Int): Double? {
```
Comment (ForNeVeR) [inline src/rider/main/kotlin/previewer/AvaloniaMessageMouseListener.kt:122]: It is unclear what this method is doing. Is it just `relativeXCoordinate`?

The same about `relSecondOrNull`, is it the same as `relativeYCoordinate`?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by rstm-sf** [comment]
> which version of Avalonia is required to test the changes?

>= 0.9.7

It seems to me that this should work with `>=0.7.0` since the last time [the file was then changed](https://github.com/AvaloniaUI/Avalonia/commit/1eda1edc045016f079685657b4939b8291ab0d57)

File: src/rider/main/kotlin/previewer/AvaloniaMessageMouseListener.kt (hunk 1 above)
Comment (rstm-sf) [reply src/rider/main/kotlin/previewer/AvaloniaMessageMouseListener.kt]: I as a result have refused these methods

**Comment by ForNeVeR** [comment]
Then I definitely don't understand something. How's it supposed to work? I've created the following XAML file:
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="clr-namespace:Avalonia.NETCoreMVVMApp2.ViewModels;assembly=Avalonia.NETCoreMVVMApp2"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="Avalonia.NETCoreMVVMApp2.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="Avalonia.NETCoreMVVMApp2">

    <Design.DataContext>
        <vm:MainWindowViewModel/>
    </Design.DataContext>

    <Button>Text</Button>

</Window>
```

Now I expect that, after this change, the `Button` will be highlighted when I move my mouse over it in Rider.

Is that true? It doesn't work for me that way.

**Comment by ForNeVeR** [comment]
It was a bug, fixed now. Thanks!

**Review by ForNeVeR — APPROVED** [review]
Thanks a lot for your contribution!

**Comment by wdcossey** [comment]
Nice @rstm-sf , I was waiting for this to be merged so I can integrate it with my updates!

**Comment by ForNeVeR** [comment]
@wdcossey I wasn't yet able to take a look at your changes, but what's the plan? Will you create a PR out of your previous changes? It would simplify the merge a bit (because if you create a PR from your repo yourself, this will, by default, grant me an ability to push into your branch and resolve merge conflicts/tiny issues I'd spot there).

Because piling up a big stack of changes in one branch without merging them upstream is asking for troubles/merge conflicts, actually.

**Comment by wdcossey** [comment]
@ForNeVeR sure I will pull the changes made by @rstm-sf and see if I can get them into my changes.

I will try have a look at it this weekend, else early next week!

**Pull request merged**